File: flutter17_mobile/lib/models/adress.dart
import 'package:json_annotation/json_annotation.dart';

part 'adress.g.dart';

@JsonSerializable()
class Adress {
  int? id;
  String? street;
  String? city;
  String? country;
  String? postalCode;
  bool? isDeleted;
  int? customerId;
  int? personId;


  Adress({
    this.id,
    this.street,
    this.city,
    this.country,
    this.postalCode,
    this.isDeleted,
    this.customerId,
    this.personId,
  });

  factory Adress.fromJson(Map<String, dynamic> json) => _$AdressFromJson(json);

  Map<String, dynamic> toJson() => _$AdressToJson(this);
}

File: angular17_web/src/app/models/adress/adress.ts
export class Adress {
  id!: number;
  street!: string;
  city!: string;
  country!: string;
  postalCode!: string;
  isDeleted!: boolean;
  customerId!: number;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.street = data.street;
    this.city = data.city;
    this.country = data.country;
    this.postalCode = data.postalCode;
    this.isDeleted = data.isDeleted;
    this.customerId = data.customerId;
  }
}
